File: output/Race2012/ListBounces.ascx.vb
Option Explicit On

Imports DotNetNuke
Imports DotNetNuke.Common
Imports DotNetNuke.Common.Globals
Imports System.Web.UI
Imports System.Collections.Generic
Imports System.Reflection
Imports DotNetNuke.Security.PortalSecurity

Namespace Crewstone Consulting ltd.Modules.Race2012

    Partial Class ListBounces
        Inherits Entities.Modules.PortalModuleBase
		Implements Entities.Modules.IActionable
        
        Protected WithEvents BouncesDataSource As System.Web.UI.WebControls.ObjectDataSource

        Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
            If HasNecessaryPermission(SecurityAccessLevel.Edit, PortalSettings, ModuleConfiguration, UserId.ToString) Then
                '
            Else
                '
            End If
        End Sub


        Protected Sub SetBouncesPortalId(ByVal sender As Object, ByVal e As System.Web.UI.WebControls.ObjectDataSourceSelectingEventArgs) Handles BouncesDataSource.Selecting
        End Sub
		
		Protected Sub ListViewBounces_ItemDataBound(ByVal sender As Object, ByVal e As System.Web.UI.WebControls.ListViewItemEventArgs)
            Dim dataItem As ListViewDataItem = CType(e.Item, ListViewDataItem)
			
            If (e.Item.ItemType = ListViewItemType.DataItem) Then
			    Dim rowView As BouncesInfo = CType(dataItem.DataItem, BouncesInfo)

            End If
        End Sub

        Protected Sub ListViewBounces_ItemCommand(ByVal sender As Object, ByVal e As System.Web.UI.WebControls.ListViewCommandEventArgs)

        End Sub
		
		#Region "Optional Interfaces"
        Public ReadOnly Property ModuleActions() As Entities.Modules.Actions.ModuleActionCollection Implements Entities.Modules.IActionable.ModuleActions
            Get
                Dim Actions As New Entities.Modules.Actions.ModuleActionCollection
                Try
                Catch exc As Exception
                    'ProcessModuleLoadException(Me, exc)
                End Try
                Return Actions
            End Get
        End Property
		#End Region
		
    End Class

End Namespace

File: output/Race2012/ListBounces.ascx.cs
using DotNetNuke;
using System.Web.UI;
using System.Collections.Generic;
using System.Reflection;
using DotNetNuke.Security.PortalSecurity;

Namespace Crewstone Consulting ltd.Modules.Race2012

    Partial Class ListBounces
        Inherits Entities.Modules.PortalModuleBase

        Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
            If HasEditPermissions(ModuleId, TabId) Then
                AddItemButton.Enabled = True
            Else
                AddItemButton.Enabled = False
            End If
        End Sub

        Protected Sub AddItemButton_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles AddItemButton.Click
            Me.FormBounces.Visible = True
        End Sub

        Protected Sub InsertItemButton_Click(ByVal sender As Object, ByVal e As System.EventArgs)
            Me.FormBounces.Visible = False
            AddItemButton.Text = "Bounces Successfully Added - Click To Add Another"
        End Sub

        Protected Sub InsertCancelButton_Click(ByVal sender As Object, ByVal e As System.EventArgs)
            Me.FormBounces.Visible = False
        End Sub

        Protected Sub GridBounces_PageIndexChanging(ByVal sender As Object, ByVal e As System.Web.UI.WebControls.GridViewPageEventArgs) Handles GridBounces.PageIndexChanging
            GridBounces.PageIndex = e.NewPageIndex
        End Sub

    End Class

End Namespace
